src/transformers/trainer.py

```diff
@@ -3201,12 +3246,24 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
 
                 The dictionary will be unpacked before being fed to the model. Most models expect the targets under the
                 argument `labels`. Check your model's documentation for all accepted arguments.
+            kwargs:
+                Additional key-word arguments to pass along for custom optimizers
 
         Return:
             `torch.Tensor`: The tensor with training loss on this batch.
         """
         model.train()
         inputs = self._prepare_inputs(inputs)
+        _is_lomo = False
+
+        if is_lomo_available():
+            from lomo_optim import AdaLomo, Lomo
+
+            _is_lomo = isinstance(_unwrap_optimizer(self.optimizer), (Lomo, AdaLomo))
+
+        # For LOMO optimizers you need to explicitly use the learnign rate
+        if _is_lomo:
+            kwargs["learning_rate"] = self._get_learning_rate()
```

Comment:
If we have the optimizer set, don't we also have `self._is_lomo_optimizer`?

Reply:
yes indeed ! changed that